inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/convolution_gpu_mmad_bfyx_to_b_fs_yx_fsv32.cl

```diff
@@ -259,21 +276,32 @@ KERNEL(convolution_mmad_bfyx_to_b_fs_yx_fsv32)(
                                 + kw * OSV * ISV;
 
                 int weights_data0 = as_int(intel_sub_group_block_read((const __global uint*)(weights + f_off)));
+#if OUTPUT_FEATURE_NUM > 16
                 int weights_data1 = as_int(intel_sub_group_block_read((const __global uint*)(weights + f_off + SUB_GROUP_SIZE*ISV)));
-
+#endif
                 PACKED_TYPE_VEC src;
 
-                __attribute__((opencl_unroll_hint(OUTPUT_X_BLOCK_SIZE)))
-                for (int i = 0; i < OUTPUT_X_BLOCK_SIZE; i++) {
-                    // src[i] = slm[kh*SLM_LINE_SIZE + (sg*OUTPUT_X_BLOCK_SIZE + i)*STRIDE_SIZE_X + kw*DILATION_SIZE_X];
-                    src[i] = line_cache[kw*DILATION_SIZE_X + STRIDE_SIZE_X*i];
-                    acc[0][i] = IMAD(acc[0][i], AS_INPUT0_TYPE_4(src[i]), as_char4(weights_data0));
-                    acc[1][i] = IMAD(acc[1][i], AS_INPUT0_TYPE_4(src[i]), as_char4(weights_data1));
-
+#if OUTPUT_Y_BLOCK_SIZE > 1
+                __attribute__((opencl_unroll_hint(OUTPUT_Y_BLOCK_SIZE)))
+                for (int yb = 0; yb < OUTPUT_Y_BLOCK_SIZE; yb++) {
+#else
+                #define yb 0
```

Comment:
Does it really improve something?

Reply:
I was surprised, but yes, it really make sense in this case